src/core/libmaven/PeakGroup.h

```diff
@@ -305,6 +305,8 @@ class PeakGroup{
 
         inline void addChildIsoWidget(const PeakGroup& child) { childrenIsoWidget.push_back(child); childrenIsoWidget.back().parent = this;   }
 
+        inline void setGroupIdForChildren() { for (auto& child : children) child.groupId = groupId; }
```

Comment:
groupId is supposed to be different for every group. Even children. MetagroupId is the id shared between the parent and children

Reply:
It was not unique until now. See here: https://github.com/ElucidataInc/ElMaven/blob/develop/src/core/libmaven/isotopeDetection.cpp#L319

Only while exporting are they assigned unque IDs and `metaGroupId` is shared. Otherwise `metaGroupId` is used as counters for the order in which a group's children were created.

Let me know if you want me to change this to the behavior you describe.

Reply:
I'd forgotten. This is fine.